pkg/provider/azure.go

```diff
@@ -532,6 +532,13 @@ func (az *Cloud) InitializeCloudFromConfig(config *Config, fromSecret bool) erro
 	az.routeUpdater = newDelayedRouteUpdater(az, routeUpdateInterval)
 	go az.routeUpdater.run()
 
+	// wait for the success first time of syncing zones
+	err = az.syncRegionZonesMap()
```

Comment:
may be not related to this PR:  is there a way to turn off `syncRegionZonesMap`? e.g. for agent node driver, it would invoke `InitializeCloudFromConfig`, while it's not necessary to run `syncRegionZonesMap` since in most cases agent node does not have permission.

Reply:
agreed, let's add a new property to allow it's to be disabled. it should only be enabled inside the controller and should be disabled by default in nodes.

Reply:
We need to know the zone info in node-manager. For example: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/67ab70534c99da15b3ec1a6c0b55d352fe638a3b/pkg/node/nodearm.go#L92

Reply:
If IMDS is not disabled (which is the common case for most scenarios except azurestack), the zone info should be got from IMDS

Reply:
I also see the same failure from unit tests, which caused unit test to run very slow until the requests timeout. We should disable it as well for most unit tests except the AZ tests

Reply:
Only enables the zone syncing in ccm now and disable it in cnm, please take a look again.

Reply:
For unit tests, I mock the zone client every time the `syncRegionZonesMap` is called, so there shouldn't be any timeouts.